### PR TITLE
Adding Windows setup script

### PR DESCRIPTION
**Pull request by bookdude13**
Makes it much easier to get started in Windows. Downloads OpenSSL, SFML, and CSFML, and sets the appropriate environment variables.

@kayla-e774 you can look at this one separate from #4/#5.

### CODE REVIEW

File: setup/zipjs.bat [hunk 1]
```diff
@@ -0,0 +1,823 @@
...
+if (! Shell.Application ) {
+	Shell.Application={};
+}
+
+if ( ! Shell.Application.ExistsFolder ) {
+	Shell.Application.ExistsFolder = function(path){
+		var ShellObj=new ActiveXObject("Shell.Application");
+		var targetObject = new Object;
+		var targetObject=ShellObj.NameSpace(path);
+		if (typeof targetObject === 'undefined' || targetObject == null ){
```
Comment (kayla-e774) [inline setup/zipjs.bat]: is the triple equals intentional?

**Comment by bookdude13** [comment]
zipjs.bat is a third party library I found, so it doesn't need a code review :)

**Comment by kayla-e774** [comment]
I was able to run windows_setup.bat and (after completing OpenSSL setup wizard) during the SFML and CSFML runs there were lines stating that "Access is denied." Also, during the appending PATH section, I got a truncation warning and a statement saying that "the directory is not empty".  I pressed a key to continue when prompted, and the entire window closed out (this report is from the second time running the script, so I could report the messages back)... I'm not sure if that's supposed to happen, but maybe there should be a warning that it could close the window, if someone is tracking the messages. I might also add a completion message, just because I felt as though there would be more (with the "press any key to conintue..." message) when the window did close out, making it feel incomplete in a way.

The entire thing looks like it ran to completion, but I'd like to know if there's a way to test this.

Also, most people running this probably wouldn't have had to try and set up their computer to run light show software before, so could that make the results of the batch script different than what most new people would encounter? (I could go through and clean everything out from past development, then run the script again if that would help to test performance.)

Also, because I've never actually looked at a batch file before: If someone ran this twice, would they get duplicates or replacements, or something else from the second run of window_setup.bat?

**Comment by bookdude13** [comment]
I can add a better end message. If it's run twice that's fine, won't change or break anything. Access denied is probably because I'm copying things to the root C drive. Will look into getting the script to run in elevated mode so that doesn't come up. Right now there's not really error handling, so if one thing goes wrong the rest of that 'section' (openssl, csfml, or sfml) will probably break (which might be the other errors you were getting).

To test if it worked, try building proton-cli or proton-runner. Will make the requested changes

**Comment by kayla-e774** [comment]
Okay, awesome! That answered all of my questions, thank you.

**Comment by kayla-e774** [comment]
Looks like I'm still missing a bin target, so proton-cli and proton-runner are still empty. Also, I'm confused about why I'm still running OpenSSL 1.0.2j with the date 26 Sep 2016 even though I went through the startup wizard for OpenSSL.

**Comment by bookdude13** [comment]
Not sure why OpenSSL is an old version. I did remember that I was testing this on a branch that wasn't merged yet :P See proton-cli PR #48. Also, when testing do `cargo clean` before `cargo build`, so everything starts rebuilding from scratch (in case built already and cached somewhere)

**Comment by bookdude13** [comment]
Changes added, ready for review

**Comment by bookdude13** [comment]
Also, after running the setup script you'll need to open a new Git Bash window for the environment variable changes to be applied

**Review by kayla-e774 — APPROVED** [review]
Everything looks good after running it on Monday!

**Pull request merged**